packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -540,6 +763,61 @@ class InteractiveViewer extends StatefulWidget {
     return closestOverall;
   }
 
+  // Convert an axis aligned Quad to a Rect.
+  //
+  // All Rects must axis aligned.
+  static Rect _axisAlignedQuadToRect(Quad quad) {
+    assert(isAxisAligned(quad));
+    double? xMin;
+    double? xMax;
+    double? yMin;
+    double? yMax;
+
+    for (final Vector3 point in <Vector3>[quad.point0, quad.point1, quad.point2, quad.point3]) {
+      if (xMin == null || point.x < xMin) {
+        xMin = point.x;
+      }
+      if (xMax == null || point.x > xMax) {
+        xMax = point.x;
+      }
+      if (yMin == null || point.y < yMin) {
+        yMin = point.y;
+      }
+      if (yMax == null || point.y > yMax) {
+        yMax = point.y;
+      }
+    }
+    return Rect.fromLTRB(xMin!, yMin!, xMax!, yMax!);
+  }
+
+  /// Returns true iff the given Quad is axis aligned.
+  @visibleForTesting
+  static bool isAxisAligned(Quad quad) {
```

Comment:
since this is only called from asserts, should we name this `debugIsAxisAligned` and be a no-op in release builds?

Reply:
Sounds good. The asserts are automatically removed in release mode, right?  Assuming you don't want me to check for kReleaseMode and return true.